llms/mlx_lm/generate.py

```diff
@@ -137,33 +132,6 @@ def setup_arg_parser():
     return parser
 
 
-def colorprint(color, s):
```

Comment:
Do you mind elaborating on what you mean by the token to Unicode mapping is many-to-mant and how this functionality it is broken? I actually make use of this functionality and find it very useful for visually incorporating token probabilities into how the generation can be 'interpreted' or even open up the door for more sophisticated things  [this](https://www.reddit.com/r/LocalLLaMA/comments/1girzia/exploring_ais_inner_alternative_thoughts_when/)

Reply:
Sometimes a single token correspond to multiple unicode characters in the output. Sometimes multiple tokens correspond to a single unicode character in the output.

Reply:
For the later case you'd have to combine the per-step probabilities to get a single color.

Reply:
You can also get the wrong result when decoding with color printing because it's not always possible to decode a chunk of correctly at each time step.

So the implementation for this needs to be rethought a bit..